nn_adapt/solving_one2n.py

```diff
@@ -0,0 +1,244 @@
+"""
+Time dependent goal-oriented error estimation
+"""
+"""
+Functions for solving problems defined by configuration
+files and performing goal-oriented error estimation.
+"""
+from firedrake import *
+from firedrake.petsc import PETSc
+from firedrake.mg.embedded import TransferManager
+from firedrake_adjoint import *
+from firedrake.adjoint import get_solve_blocks
+from pyroteus.error_estimation import get_dwr_indicator
+import abc
+from time import perf_counter
+
+import matplotlib.pyplot as plt
+
+
+tm = TransferManager()
+
+
+class Solver(abc.ABC):
```

Comment:
This is identical to the base class found in `solving.py`. Why not just import it from there?